app/plugin_requests.go

```diff
@@ -135,6 +137,9 @@ func (a *App) servePluginRequest(w http.ResponseWriter, r *http.Request, handler
 		token = r.URL.Query().Get("access_token")
 	}
 
+	// Mattermost-Plugin-ID can only be set by inter-plugin requests
+	r.Header.Del("Mattermost-Plugin-ID")
```

Comment:
I am not sure if we should follow the CanonicalHeaderKey pattern here, i.e: 
`r.Header.Del("Mattermost-Plugin-Id")`

Reply:
`Mattermost-Plugin-ID` is the header name that the GitHub and autolink plugins use. In order for the fix to also apply to the GitHub we need to stick with the same name.